padl/transforms.py

```diff
@@ -1490,6 +1500,64 @@ def __init__(self, transforms: Iterable[Transform], call_info: inspector.CallInf
         self.pd_keys = self._pd_get_keys(self.transforms)
         self._pd_output_format = namedtuple('namedtuple', self.pd_keys)
 
+    def _pd_get_splits(self, input_components=0) -> Tuple[Union[int, List],
+                                                          Tuple[Transform, Transform, Transform]]:
+        """See the docstring of :meth:`Transform._pd_splits` for more details.
+
+        A rollout splits into:
+            - the rollout of its sub-transform' first non-Identity stage
+            - the parallel of its sub-transform' remaining stages
+
+        To see why the first non-Identity stage is a rollout whereas the remaining stages are
+        parallel, note that the first non-Identity stage splits the pipeline and it remains
+        split for the rest:
+
+            Case 1:     Case 2:     Case 3:
+            pre + pre   Identity    Identity
+            for / for   for + for   Identity
+            pos / pos   pos / pos   pos + pos
+
+        The *output_components* are the list of output components of the sub-transforms.
+        """
+        if self._pd_splits is None or self._pd_splits[0][0] != input_components:
+            # TODO Should Rollout be able to accept input_components consisting of a list of ints?
+            assert isinstance(input_components, int), 'Rollout can only accept one input'
```

Comment:
@wuhu Should Rollout be able to accept input_components consisting of a list of ints? I believe if it's a list then there are multiple "pipes" and `Rollout` is not valid in that case, right?

Reply:
It's still valid, it just means that those multiple pipes get rolled out.

For example, this
```
(a / b / c) >> (x + y + z)
```
would cause the Rollout `(x + y + z)` to get a list [0, 0, 0] (or other numbers, depending on what happens in a, b, c). This is because the output of (a / b / c) - the input of (x, y, z) has three parts. That input would be passed to each of the three transforms a, b and c. Thus, the whole list would be passed to all three, x, y and z.

Reply:
Okay I understand. I'll remove the `assert` and add a test for this case, to make sure things work as expected.